core/src/test/java/com/linecorp/armeria/server/HttpServerUpgradeHandlerTest.java

```diff
@@ -195,4 +212,87 @@ void upgradeFail() {
         assertThat((Object) channel.readOutbound()).isNull();
         channel.finishAndReleaseAll();
     }
+
+    @Test
+    void upgradeFailWithBadRequest() {
+        final HttpServerCodec httpServerCodec = new HttpServerCodec();
```

Comment:
These test suites are previously copied from Netty. I don't see we need to use `UpgradeCodecFactory` to test duplicate `HTTP2-Settings`.
How about 1) starting an Armeria server using `ServerExtension` and 2) sending a raw HTTP/1 upgrade using a JDK's `Socket`?
https://github.com/line/armeria/blob/3491204f94311ab94eb54f224064f14c33896466/core/src/test/java/com/linecorp/armeria/server/ServerMaxConnectionAgeTest.java#L177-L180